odoo/service/model.py

```diff
@@ -52,8 +55,8 @@ def get_public_method(model: BaseModel, name: str):
     for mro_cls in cls.mro():
         if not (cla_method := getattr(mro_cls, name, None)):
             continue
-        if name.startswith('_') or getattr(cla_method, '_api_private', False):
```

Comment:
no need to test `name.startswith('_')` for every class in the MRO, testing it once is enough (and faster)